### PR TITLE
PML-382: occupancy readout fix for renormalization

### PR DESCRIPTION
**Pull request by CassNot**
## Summary

This was a bug flagged by Claude review on 0.4.0
**Occupancy readout was silently renormalizing when `probs()` did not** : there was an inconsistency in how probability is handled between the occupancy readout path and the non-occupancy `probs()` path when photon loss or detector  occurs upstream.
- **Issue**: `_OccupancyReadout.forward` was rescaling grouped probabilities to sum=1 after grouping
- **Non-occupancy path**: `DistributionStrategy.process` (in `strategies.py:147-158`) never renormalizes
- **Impact**: Under photon loss or detector mass-loss, the two paths diverge in total probability mass with **no signal to the user**

## Related Issue
   
PML-382

## Type of change
- [x] Bug fix
- [ ] New feature
- [ ] Documentation update
- [ ] Refactor / Cleanup
- [ ] Performance improvement
- [ ] CI / Build / Tooling
- [ ] Breaking change (requires migration notes)

## Proposed changes

I am converting occupancy grouping to a **mass-preserving column sum**:

1. **Removed** the normalization step that rescaled grouped probabilities back to sum=1
2. **Documented** the mass-preserving behavior explicitly in docstrings
3. **Aligned** occupancy grouping semantics with the non-occupancy `probs()` path
### Before (Buggy)
```python
# Input: probabilities sum to 0.5 (due to upstream photon loss)
probabilities = torch.tensor([[0.10, 0.20, 0.15, 0.05]])  # sum = 0.5

readout = _OccupancyReadout([(0, 0), (1, 0), (2, 0), (0, 1)])
grouped = readout(probabilities)

grouped.sum()  # Output: 1.0 (silently rescaled!)
```

### After (Fixed)
```python
# Same input: probabilities sum to 0.5
probabilities = torch.tensor([[0.10, 0.20, 0.15, 0.05]])  # sum = 0.5

readout = _OccupancyReadout([(0, 0), (1, 0), (2, 0), (0, 1)])
grouped = readout(probabilities)

grouped.sum()  # Output: 0.5 (mass is preserved!)
```

## Testing

Added comprehensive test suite in `tests/measurement/test_readouts.py` with 5 new test cases:

1. **`test_lossy_input_mass_is_preserved_not_renormalized`**: Verifies that sub-1 total mass is preserved (the core bug fix)
2. **`test_grouped_values_match_column_sum_reference`**: Ensures grouping is a plain column sum with no division step
3. **`test_fully_normalized_input_still_sums_to_one`**: Sanity check that normalized inputs stay normalized
4. **`test_1d_lossy_input_mass_is_preserved`**: Verifies the unbatched (1-D) code path also preserves mass
5. **`test_zero_mass_row_stays_zero`**: Edge case: fully-lost rows (all zeros) don't cause divide-by-zero

## Documentation
- [ ] User docs updated (Sphinx)
- [ ] Examples / notebooks updated
- [x] Docstrings updated
- [ ] Updated the API

## Checklist
- [x] PR title includes Jira issue key (e.g., PML-126)
- [x] "Related Jira ticket" section includes the Jira issue key (no URL)
- [x] Code formatted (ruff format)
- [x] Lint passes (ruff)
- [x] Static typing passes (mypy) if applicable
- [x] Unit tests added/updated (pytest)
- [x] Tests pass locally (pytest)
- [x] Tests pass on GPU (pytest)
- [x] Test coverage not decreased significantly
- [x] Docs build locally if affected (sphinx)
- [x] With this command: `SPHINXOPTS="-W --keep-going -n" make -C docs clean html` the docs are built without any warning or errors.
- [x] New public classes/methods/packages are added in the API following the methodology presented in other files.
- [x] Dependencies updated (if needed) and pinned appropriately
- [x] PR description explains what changed and how to validate it

### CODE REVIEW

**Comment by CassNot** [comment]
@claude review

**Comment by CassNot** [comment]
@claude review

**Review by ben9871 — APPROVED** [review]
seems logical. makes sure occupancy obeys the same precedent as probs and doesn't renormalize. tests and helpers done according to Claude feedback
